backend/Squirrel.Core/Squirrel.Core.BLL/Services/UserService.cs

```diff
@@ -53,6 +58,66 @@ public async Task<User> CreateUserAsync(UserRegisterDto userDto, bool isGoogleAu
         return createdUser;
     }
 
+    public async Task<UserDto> UpdateUserAsync(UpdateUserNamesDTO updateUserDTO)
+    {
+        var userEntity = await GetUserByIdInternal(updateUserDTO.Id);
+
+        userEntity.Username = updateUserDTO.Username;
+        userEntity.FirstName = updateUserDTO.FirstName;
+        userEntity.LastName = updateUserDTO.LastName;
+
+        _context.Users.Update(userEntity);
+        await _context.SaveChangesAsync();
+
+        return _mapper.Map<UserDto>(userEntity);
+    }
+
+    public async Task ChangePasswordAsync(UpdateUserPasswordDTO changePasswordDTO)
+    {
+        var userEntity = await GetUserByIdInternal(changePasswordDTO.Id);
```

Comment:
Can you please load id from token, as this way it's possible to update any user (communicate with Oleg not to repeat work)

Reply:
updated to use `_userIdGetter.GetCurrentUserId()`